### PR TITLE
Benji/ios 565 add unread header to conversation list

### PR DESCRIPTION
**Pull request by Benjamindodgson**

### CODE REVIEW

File: iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift [hunk 1]
```diff
@@ -55,7 +55,7 @@ class ConversationMessagesCell: UICollectionViewCell, ConversationUIStateSettabl
     /// A reference to the current task that scrolls to a specific message
     private var scrollToMessageTask: Task<Void, Never>?
     /// If true we should scroll to the last item in the collection in layout subviews.
-    private var scrollToFirstUnreadIfNeccessary: Bool = false
+    private var scrollToFirstMessageIfNeccessary: Bool = false
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift]: No longer scrolls to first unread.

File: iOS/Flows/Conversation/Input/InputHandlerCoordinator.swift [hunk 2]
```diff
@@ -11,11 +11,13 @@ import PhotosUI
 import Photos
 import Localization
 import Lightbox
+import StreamChat
 
 protocol SwipeableInputControllerHandler where Self: ViewController {
     var messageCellDelegate: MesssageCellDelegate? { get set }
     var swipeableVC: SwipeableInputAccessoryViewController { get }
     func updateUI(for state: ConversationUIState, forceLayout: Bool)
+    func scrollToConversation(with cid: ConversationId, messageId: MessageId?) async
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Input/InputHandlerCoordinator.swift:20]: All need this.

File: iOS/Flows/Conversation/Input/InputHandlerCoordinator.swift [hunk 3]
```diff
@@ -62,6 +64,23 @@ class InputHandlerCoordinator<Result>: PresentableCoordinator<Result>,
...
+    
+    func scrollToUnreadMessage() {
+        guard let conversation = ConversationsManager.shared.activeConversation,
+                let unreadMessage = conversation.messages.first(where: { message in
+            return !message.isFromCurrentUser && !message.isConsumedByMe
+        }) else { return }
+        
+        Task.onMainActorAsync {
+            await Task.sleep(seconds: 0.1)
+            await self.inputHandlerViewController.scrollToConversation(with: conversation.cid,
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Input/InputHandlerCoordinator.swift:81]: adding allows for this at the base.

File: iOS/_Stream Temp/Conversation+Extensions.swift [hunk 4]
```diff
@@ -113,7 +113,7 @@ extension Conversation: MessageSequence {
     var totalUnread: Int {
         var total: Int = 0
         self.messages.forEach { message in
-            if !message.isFromCurrentUser, !message.isConsumedByMe {
+            if !message.isFromCurrentUser, !message.isConsumedByMe, !message.isDeleted {
```
Comment (Benjamindodgson) [inline iOS/_Stream Temp/Conversation+Extensions.swift:116]: Was getting unread message count for deleted messages.

File: iOS/Flows/Conversation/Input/UnreadMessagesCounter.swift [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            self.counter.setValue(Float(conversation.totalUnread))
+        }.store(in: &self.cancellables)
+        
+        ConversationsManager.shared.$reactionEvent.mainSink { event in
+            guard let reactionEvent = event as? ReactionNewEvent,
+                  let conversation = ConversationsManager.shared.activeConversation,
+                  reactionEvent.cid == conversation.cid else { return }
+            
+            self.counter.setValue(Float(conversation.totalUnread))
+        }.store(in: &self.cancellables)
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Input/UnreadMessagesCounter.swift]: Looks like a memory leak. Unowned self needed.

**Review by MartinJibber — CHANGES_REQUESTED** [review]
Looks good. Just one change needed

File: iOS/Flows/Conversation/Input/UnreadMessagesCounter.swift (hunk 5 above)
Comment (Benjamindodgson) [reply iOS/Flows/Conversation/Input/UnreadMessagesCounter.swift]: ah yes. ty

**Pull request merged**